File: proof-of-concept/spring-mybatis-poc/src/main/java/shell/App1.java
package shell;

import model.ManagerCuentas;

public class App1 {

  public static void main(final String[] args) {
    App1 app1 = new App1();
    app1.transferir();
  }

  private void transferir() {

    ManagerCuentas managerCuentas = SpringBeanRetriever.getBean("managerCuentas");

    try {

      String desdeCta = "C02";
      String haciaCta = "A08";
      Integer monto = 5000;

      System.out.println(" ");
      System.out.println("Transfiriendo $" + monto + " - desde " + desdeCta + " - hacia " + haciaCta + " ...");

      managerCuentas.transferir(desdeCta, haciaCta, monto);

      System.out.println(" ");
      System.out.println("Transferencia exitosa.");

    } catch (RuntimeException e) {
      System.out.println(" ");
      System.out.println("Transferencia fallida (motivo: " + e.getClass().getSimpleName() + ").");
    }

    System.out.println(" ");
    managerCuentas.listarCuentas();

  }

}

File: tests/attic/proof-of-concept/spring-mybatis-kotlin-poc/src/main/kotlin/shell/App1.kt
package shell

import model.ManagerCuentas;

fun main(args: Array<String>) {
  println("[Starting]")
  transferir()
  println("[Finished]")
}

fun transferir() {

  val managerCuentas: ManagerCuentas = SpringBeanRetriever.getBean("managerCuentas") as ManagerCuentas

  try {

    val desdeCta: String = "C02"
    val haciaCta: String = "A08"
    val monto: Int = 5000

    println(" ")
    managerCuentas.listarCuentas()

    println(" ")
    println("Transfiriendo $" + monto + " - desde " + desdeCta + " - hacia " + haciaCta + " ...")

    managerCuentas.transferir(desdeCta, haciaCta, monto)

    println(" ")
    println("Transferencia exitosa.")

  } catch (e: RuntimeException) {
    println(" ")
    try {
      println("Transferencia fallida")
      println(" - motivo: " + e::class.simpleName)
    } catch (e2: Exception) {
      println("Exception")
      println("Exception: " + e::class.simpleName)
    }
  }

  println(" ")
  managerCuentas.listarCuentasConNombre()

}